xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ObjectAddForm.java

```diff
@@ -47,7 +47,12 @@ public void setClassName(String className)
     @Override
     public void readRequest()
     {
-        setClassName(getRequest().getParameter("classname"));
+        // We need to support both 'classname' and 'className' since an inconsistency has been introduced.
```

Comment:
I'd rephrase that a bit saying that we keep "className" for backward compatibility and that we're updating to "classname" to be consistent with the parameter name used for Class Name for other actions.